iOS Client/Store/MBPodcastsStore.swift

```diff
@@ -8,38 +8,51 @@
 
 import Foundation
 import CoreData
+import PromiseKit
 
 
 class MBPodcastsStore {
     
     let client: MBClient
     let fileHelper: FileHelper
+    let dateFormatter: DateFormatter = {
+        let d = DateFormatter()
+        d.dateFormat = "E, d MMM yyyy HH:mm:ss Z"
+        return d
+    }()
     
     init() {
         client = MBClient()
         fileHelper = FileHelper()
     }
     
-    func syncPodcasts(completion: @escaping ([MBPodcast]?, Error?) -> Void) {
-        self.client.getPodcastsWithCompletion { data, error in
-            if error == nil, let data = data {
-                do {
-                    // We got some data now parse
-                    print("fetched podcasts from server")
-                    let parser = XMLParser(data: data)
-                    let xmlParserDelegate = PodcastXMLParsingDelegate()
-                    parser.delegate = xmlParserDelegate
-                    parser.parse()
-                    let podcasts = xmlParserDelegate.podcasts
-                    try self.fileHelper.save(podcasts, forPath: "podcasts")
-                    completion(podcasts, nil)
-                } catch let error {
-                    completion(nil, error)
+    func syncPodcasts() -> Promise<[DisplayablePodcast]> {
+        let streams: [MBClient.PodcastStream] = [.pz, .mockingPulpit, .mockingCast]
+        let requests = streams.map{ self.client.getPodcasts(for: $0)}
+        return firstly {
+            when(resolved: requests)
+        }.then { responses -> Promise<[DisplayablePodcast]> in
+            var podcasts: [DisplayablePodcast] = []
+            for (indx, response) in responses.enumerated() {
+                if case .fulfilled(let newCasts) = response {
+                    let displayCasts = newCasts.flatMap { podcast -> DisplayablePodcast? in
+                        guard let dateString = podcast.pubDate, let date = self.dateFormatter.date(from: dateString) else {
+                            return nil
+                        }
+                        return DisplayablePodcast(author: podcast.author,
```

Comment:
Hmm, I go back and forth. There may be some value in splitting our DTO (data transfer object) model from our domain model . . .

Reply:
My original intent for splitting it was for the images, but now that we are just storing the name of the image I think it might be ok to consolidate to one model